src/app.jsx

```diff
@@ -607,29 +608,77 @@ const NavigatorCardBody = ({
         );
     };
 
+    const EmptyDirectoryMessage = () => {
+        return (
+            <EmptyState>
+                <EmptyStateBody>
+                    Directory is empty
+                </EmptyStateBody>
+            </EmptyState>
+        );
+    };
+
     if (loadingFiles)
         return (
             <Flex justifyContent={{ default: "justifyContentCenter" }}>
                 <Spinner />
             </Flex>
         );
+
     if (isGrid) {
         return (
             <CardBody onClick={resetSelected} id="navigator-card-body">
-                <Gallery id="folder-view">
-                    {sortedFiles.map(file => <Item file={file} key={file.name} />)}
-                </Gallery>
+                {sortedFiles.length === 0
+                    ? (
+                        <EmptyDirectoryMessage />
+                    )
+                    : (
+                        isGrid
+                            ? (
+                                <Gallery id="folder-view">
+                                    {sortedFiles.map(file => <Item file={file} key={file.name} />)}
+                                </Gallery>
+                            )
+                            : (
+                                <ListingTable
+                                  id="folder-view"
+                                  className="pf-m-no-border-rows"
+                                  variant="compact"
+                                  columns={[_("Name")]}
+                                  rows={sortedFiles.map(file => (
+                                      { columns: [{ title: <Item file={file} key={file.name} /> }] }))}
+                                />
+                            )
```

Comment:
This makes not a lot of sense, it is already behind a `if (isGrid)` on line 627.